extension/dynamicconfig/model/pattern.go

```diff
@@ -0,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Contains common models for the dynamic config service. The corresponding
+// Proto() methods convert the model representation to a usable struct for
+// protobuf marshalling.
+
+package model
+
+import (
+	pb "github.com/open-telemetry/opentelemetry-proto/gen/go/collector/dynamicconfig/v1"
+	"hash/fnv"
+)
+
+type Pattern struct {
```

Comment:
Can you add a comment saying only of them is not empty? And Equals has the precedence?

Reply:
Comment added.

I've altered the models logic so that the `Proto()` methods return errors as well as their original value. Since the user is free to build the models as they see fit, it seemed like a necessary addition to add some sort of error checking -- either with a `New()` type method, or with `Proto()`. I've opted for the latter to preserve the freedom to initialize the model structs using literals. In this way, the logic to check that only one of the fields is used in Pattern is enforced using the error.